src/tasks/netcore/NetCoreTaskHelper.ts

```diff
@@ -287,6 +298,44 @@ export class NetCoreTaskHelper implements TaskHelper {
 
         return volumes;
     }
+    /**
+     * @see {@link https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container} for further information
+     */
+    private async buildWithDotnetSdk(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition): Promise<void> {
+        const configuration = buildDefinition.netCore?.configuration ? `-c ${buildDefinition.netCore?.configuration} ` : '';
+        const publishFlag = NetCoreTaskHelper.isWebApp ? '-p:PublishProfile=DefaultContainer' : '/t:PublishContainer';
+        const [runtimeOs, runtimeArch] = this.getOsAndArch(buildDefinition);
+        const imageNameInfo = parseDockerLikeImageName(buildDefinition.dockerBuild.tag);
+        const sdkBuildCommand = `dotnet publish --os ${runtimeOs} --arch ${runtimeArch} ${publishFlag} ${configuration}-p:ContainerImageName=${imageNameInfo.image} -p:ContainerImageTag=${imageNameInfo.tag}`;

```

Comment:
I'd suggest using the commandLineBuilder methods to make this code easier to maintain/debug.

Something like:

const sdkBuildCommand = composeArgs(
  withArg('dotnet', 'publish'),
  withNamedArg('-c', buildDefinition.netCore?.configuration),
  ...
)();

await context.terminal.execAsyncInTerminal(
  Shell.getShellOrDefault().quote(sdkBuildCommand),
  ...
);